CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  ErrorType.swift
+//  
+//
+//  Created by Krzysztof Kryniecki on 24/11/2022.
+//
+
+import Foundation
+import GiniBankAPILibrary
+
+@objc public enum ErrorType: Int {
+    case connection
+    case request
+    case uploadIssue
+    case serverError
+    case authentication
+    case unexpected
+    case importError
+
+    public init(error: GiniError) {
+        switch error {
+        case .unauthorized(_, _), .invalidCredentials, .keychainError:
+            self = .authentication
+        case .noResponse:
+            self = .connection
+        case .notAcceptable(let response, _), .tooManyRequests(let response, _),
+                .parseError(_, let response, _), .badRequest(let response, _), .notFound(let response, _):
+            if let status = response?.statusCode {
+                switch status {
+                case 400, 402 ... 499:
+                    self = .request
+                case 401:
+                    self = .authentication
+                case let code where code >= 500:
+                    self = .serverError
+                default:
+                    self = .unexpected
+                }
+            } else {
+                self = .serverError
+            }
+        default:
+            self = .unexpected
+        }
+    }
+
+    func iconName() -> String {
```

Comment:
[Suggestion] Declaring these as variables not functions would reduce complexity

Reply:
`
var iconName: String {

        switch self {
        case .connection:
            return "errorCloud"
        case .request:
            return "alertTriangle"
        case .authentication:
            return "errorAuth"
        case .serverError:
            return "errorGlobe"
        case .unexpected:
            return "alertTriangle"
        case .uploadIssue:
            return "errorUpload"
        case .importError:
            return "alertTriangle"
        }
    }`

Reply:
hmm @vizaknaidavid  What kind of complexity ? I see that the difference would be just the way of invoking it without '()' parenthesis.